ferrocene/doc/specification/src/types-and-traits.rst

```diff
@@ -2914,6 +2892,16 @@ Trait and Lifetime Bounds
    ForGenericParameterList ::=
        $$for$$ GenericParameterList
 
+   UseCaptures ::=
```

Comment:
`TypeBound` also needs to be updated to include `UseCapture` as one of its variants. Then we need to semantically forbid these bounds from appearing in type bound lists, unless the list is from an anonymous return type

Reply:
I assume this is what rust-lang/reference refers to as Abstract return type (`-> impl Trat`)